app/controllers/api/configuration_script_sources_controller.rb

```diff
@@ -16,20 +16,15 @@ def delete_resource_main_action(type, config_script_src, _data)
       {:task_id => config_script_src.delete_in_provider_queue}
     end
 
-    def create_resource(_type, _id, data)
-      validate_attrs(data)
+    def create_resource(type, _id, data)
+      # TODO: convert configuration script api to (also?) support standard `ems_id` key
       manager_id = parse_id(data['manager_resource'], :providers)
-      raise 'Must specify a valid manager_resource href or id' unless manager_id
-      manager = resource_search(manager_id, :providers)
```

Comment:
@agrare ^?

Reply:
I think part of the reason it's an oddball is because it's use as the git resource for embedded ansible, so it's not a "normal" provider, per se.  Not sure if any "real" providers use it though.

Reply:
```ruby
[2] pry(main)> ConfigurationScriptSource.descendants.map(&:name)
=> ["ManageIQ::Providers::AutomationManager::ConfigurationScriptSource",
 "ManageIQ::Providers::EmbeddedAutomationManager::ConfigurationScriptSource",
 "ManageIQ::Providers::EmbeddedAnsible::AutomationManager::ConfigurationScriptSource",
 "ManageIQ::Providers::ExternalAutomationManager::ConfigurationScriptSource",
 "ManageIQ::Providers::AnsibleTower::AutomationManager::ConfigurationScriptSource"]
```

Reply:
Other than not having `supports?(:create)` and calling it a manager_resource instead of an ems_id (maybe b/c of the parent `Provider` that automation_managers have? this doesn't look that odd ball to me

Reply:
I was able to get it close to the code, but you notice that we're using a different ems_id and it kinda fetches it from a nested hash.

cool. as long as it works

Reply:
Yeah we could do a follow-up to have this match the rest of these types of create methods, even among the "normal" looking ones we have a mix of `klass.create_queue(:ems_id => )` and `ems.create_queue`

Reply:
ok to accept `manager_resource || ems_id`, or does `ems_id` not logically make sense?

This would allow us to migrate to the common create pattern

Reply:
We could change the parameter to ems_id but that would be a "breaking" API change

Reply:
I like the idea of supporting both and breaking it at some future date

Reply:
:+1: Yeah you could do `data['ems_id'] ||= data.delete('manager_resource')` prior to calling the standard `create_ems_resource` that way either would work.